### ISSUE TITLE
Document how to manipulate data sourced with NCBI SRA tools

### ISSUE THREAD

**Issue by jennaj** [galaxyproject/galaxy-hub#216]
Page is started here, needs more work: https://galaxyproject.org/support/ncbi-sra-fastq/

Ideas:

- Cover both BAM and Fastq manipulation/prep?
- Add highlights from a tutorial that covers this and link that in? Do any already exist?
- Create a published workflow or use an existing one that splits the data? 

@MoHeydarian Would you like to work with me on this? We talked about expanding the help, but I wasn't sure where we were with other docs (including teaching materials) that might include examples. In short, decide on our "Best practices" for this data source and share with users. We could move this under /learn instead of /support if you'd prefer.

**Comment by MoHeydarian** [comment]
@jennaj Yes, I would like to work with you on this! I've got a short blurb of how to use NCBI SRA tools to import data as collections that I haven't found a home for, this is the perfect place for it.

**Comment by jennaj** [comment]
Are we good with https://galaxyproject.org/support/ncbi-sra-fastq/ now? Close this out?

**Comment by jennaj** [comment]
@MoHeydarian 

The support FAQ helps but is not quite enough yet. https://galaxyproject.org/support/ncbi-sra-fastq/#ncbi-sra-sourced-fastq-data

Many users still run into problems around this and don't quite understand what is going wrong or how to fix it. Downstream tools simply error in confusing ways (that wouldn't necessarily error if run line-command). Example: In Galaxy, FastQC can handle the differing qual `+` description, but Trimmomatic/others cannot.

The data doesn't route through the Upload tool, so doing cleanup for the user at that point is not an option (and the de-interlacing part is not always needed/wanted anyway). Plus, compressed fq complicates it all...

Ideas:

__Tool solutions__
* New tool request for IUC, named something like "NormalizeFastq"? Have it include default "cleanup" for qual score `+` lines, plus an option to de-interlace (with a user specified "split" terms?). Point would be to do everything in one tools to make it easier, even if other methods exist (the same way "NormalizeFasta" simplifies common cleanup). 
* Have tools captured these data issues better and reported clearly what the problem is and what to do to fix it, directly in error message displayed in the red dataset and when bug icon is clicked on.
 * **Upgrade QC tools like Trimmomatic to include an option to do the `+` cleanup at that step (default toggle to "yes", or maybe make it more error-proof and always do the cleanup). AND upgraded tool to accept interlaced fastq, and output de-interlaced as a user defined option. Both would be a big win, imo.** This is my favorite option, after thinking about all the other ways we could address this -- is the least complicated.


__GTN/Tutorial solutions__
* Get this added to an existing tutorial.
* Or maybe better, create a standalone micro-tuto under Data Mani category. 
* This is my least favorite option - Galaxy is easier to use when tools anticipate common, problematic data formats and handle/fix it, directly. 

 __GUI solutions__
* Consider adding in a tooltip or metadata warning in the GUI with an FAQ/Tuto link, for data imported from EBI (and NCBI in some cases). 
* Not sure about the utility of having an "interlaced fastq" datatype, but could potentially help, the same way that creating Bam: sorted/not datatype breakout helped.
* Avoid the the issue/extra workarounds and have Trimmomatic do the data fix-ups.

 __Tool "help section" solutions__
  * No way to add info to the EBI tool help, since the EBI SRA get data method doesn't have an intermediate tool form (goes directly to the data provider, and that process should remain as-is for a few reasons, imo). 
  * NCBI SRA fastq dump does have a tool form, and including an additional option or more help there might be useful ("how-to" for de-interlacing fastq data). ??
  * I still see many users attempting to use the Splitter tool when troubleshooting on their own, for either source or their own data, which doesn't do the appropriate manipulation of course. Maybe improve the help there to avoid confusion, link in FAQ/tuto, etc. 
* Or, again, avoid the issue with Trimmomatic changes.

Thoughts or is this being addressed in some other way already?

**Comment by jmchilton** [comment]
@jennaj and I discussed this and decided adding an option to strip the relevant lines from the FASTQ file in the fastq-groomer tool would be a good start.

- These are the lines: https://galaxyproject.org/support/ncbi-sra-fastq/#ncbi-sra-sourced-fastq-data
- This is the source code for the package the fastq groomer tool utilizes (https://github.com/galaxyproject/sequence_utils).

**Comment by mvdbeek** [comment]
Another suggestion would be to raise these points directly with the ncbi-toolkit devs,
and to improve the sra-tools wrapper at the IUC.
There are options in fastq-dump to avoid interlaced output, and as far as I can tell this is not an issue when downloading fastqs through the ENA for instance, but we'd need some concrete examples.

**Comment by mvdbeek** [comment]
Also fastq-dump is being superseded by https://github.com/ncbi/sra-tools/wiki/HowTo:-fasterq-dump, which seems to have much saner defaults.

**Comment by jennaj** [comment]
I have example accessions that have problems at ENA (the NCBI version never has these). I'll put them in a clean test history and share it back, with tags added for the root problem. Then can decide how to address, and where. Thanks!

**Comment by jennaj** [comment]
Related:

https://github.com/galaxyproject/tools-iuc/issues/1099
https://github.com/galaxyproject/tools-iuc/issues/970

**Comment by jennaj** [comment]
Well, it looks like @pvanheus made a commit to fix this in the tool wrapper itself in 2017 !!!(https://github.com/galaxyproject/tools-iuc/commit/9d72ba494608310e75d5ddb68bd885dc25dcb9ff#diff-2ba4c936db7184d75e82fb3d32430400)

That looks like it didn't get added to the updated tool in the MTS till Dec 2018, version `2.8.1.3`. Or the prior version `2.8.1.2` didn't bump versioning with the fix, and we missed catching the bug fix to update it on Main.

Not sure when `2.8.1.3` was updated on Main, but that did happen sometime in the last few months and the problem is gone! People must still have older/staged data around on Main. Reported issues will die off over time -- but just on Main.

This history has a comparison between version `2.8.1.2` (with qual identifier problem, from a prior run) and version `2.8.1.3` (without qual identifier problem, run today): https://usegalaxy.org:/u/jen/h/test-ncbi-sra-fastq-content

**I still think this is worth adding to the Groomer tool.** Only the new version is at Main (older versions are not hidden, so cannot be used/rerun), but older versions of the tool are still available in tool versions lists at EU & AU (see below). Plus older/other data with formatting issues might come up still -- might as well have the groomer clean this up, along with other things it does. ping @ic4f 

cc @bgruening @Slugger70 there are older versions of the tool at the EU and AU public servers. Main just has the most current version. Just an FYI if you guys are seeing reported errors that trace back to this.

This accession tested is also interlaced/interleaved. That is what the linked issues are about (can't actually split within the tool under advanced options). I did not retest that yet, but will, and put the results back in those tickets. Maybe can be closed out, maybe not.

----

Scope creep: standardizing the nomenclature for "interlaced" versus "interleaved" across all tools/help/gui would help. I'll make a ticket, has been bothering me for a while.

**Comment by ic4f** [comment]
@jennaj :+1: on standardizing the nomenclature for "interlaced" versus "interleaved" . Took me quite some time to research both(!) terms..

**Comment by jennaj** [comment]
FastqGroomer update PR: https://github.com/galaxyproject/usegalaxy-playbook/issues/237

Is in tool install (hold) list but we can combine/clear both once all done.

Part of this will be me updating the NCBI data source FAQs. I'll also review the tool new help text, in context when I try to "break it" in common ways that users do :)
